### PR TITLE
added RarArchive extension

### PR DESCRIPTION
**Pull request by AmsTaFFix**

### CODE REVIEW

File: rar.php [hunk 1]
```diff
@@ -0,0 +1,620 @@
+<?php
+
+/**
+ * This class represents a RAR archive, which may be formed by several volumes (parts) and which contains a number of
+ * RAR entries (i.e., files, directories and other special objects such as symbolic links).
+ *
+ * Objects of this class can be traversed, yielding the entries stored in the respective RAR archive.
+ * Those entries can also be obtained through {@see \RarArchive::getEntry} and {@see \RarArchive::getEntries}.
+ *
+ * @link http://php.net/manual/ru/class.rararchive.php
```
Comment (avsej) [inline rar.php]: it is better strip locale from url, and rely on user defaults. PHP.net will redirect to correct locale
http://php.net/manual/class.rararchive.php

**Comment by neuro159** [comment]
can you please move the stub into its separate folder (==module name)?
we're in the midst of the cleanup to simplify control over modules
(UI allowing per-project switching is also added)

File: rar.php (hunk 1 above)
Comment (AmsTaFFix) [inline rar.php]: my mistake, i will change

**Comment by 6ecuk** [comment]
Hi, if  any plans for merge this request in master  ? I need RarArchive stub in PhpStorm

**Comment by andrey-sokolov** [comment]
Merged https://github.com/JetBrains/phpstorm-stubs/pull/234

**Pull request closed without merging**